specs/Standards/NonFungibleToken/Payout.md

```diff
@@ -80,18 +83,27 @@ pub trait Payouts{
     receiver_id: AccountId,
     token_id: String,
     approval_id: u64,
+    memo: Option<String>,
     balance: U128,
-    max_len_payout: u32,
-  ) -> Payout{
+    max_len_payout: Option<u32>,
+  ) -> Payout {
     assert_one_yocto();
     let payout = self.nft_payout(token_id, balance);
-    self.nft_transfer(receiver_id, token_id, approval_id);
+    self.nft_transfer(receiver_id, token_id, approval_id, memo);
     payout
   }
 }
 ```
 
-Note that NFT and financial contracts will vary in implementation. This means that some extra CPU cycles may occur in one NFT contract and not another. Furthermore, a financial contract may accept fungible tokens, native NEAR, or another entity as payment. Transferring native NEAR tokens is less expensive in gas than sending fungible tokens. For these reasons, the maximum length of payouts may vary according to the customization of the smart contracts.
+## Fallback on error
+
+In the case where either the `max_len_payout` causes a panic, or a malformed `Payout` is returned, the caller contract should transfer all funds to the token owner.
+
+## Potential pitfalls
+
+The payout must include all accounts that should receive funds. Thus the token owner must be included.
```

Comment:
I wouldn't force the token owner in here. What if an NFT contract wanted an NFT where the owner didn't get any funds?

Reply:
This should be reworded.  The point is that it's a pitfall to assume that you only need to specify those who get royalties and the market will send the rest to the owner.  I agree with the payout being the list of all who should receive funds.